### PR TITLE
Bratseth/optimization on sddocname and recall

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Comment by bratseth** [comment]
Test failure is unrelated.

File: container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java [hunk 1]
```diff
@@ -200,6 +204,21 @@ private static void removeOtherNonrankedChildren(CompositeItem parent, int index
                 parent.removeItem(i);
         }
     }
+
+    private static boolean isRanked(Item item) {
+        if (item instanceof CompositeItem) {
+            boolean isRanked = false;
+            for (Item child : ((CompositeItem)item).items())
+                isRanked |= isRanked(child);
```
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: if (isRanked(child)) return true;
No need to iterate all children if one is unhappy.

File: container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java [hunk 2]
```diff
@@ -200,6 +204,21 @@ private static void removeOtherNonrankedChildren(CompositeItem parent, int index
...
+            return isRanked;
```
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: return false;

**Review by baldersheim — APPROVED** [review]

File: container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java (hunk 2 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: ok, done

**Pull request merged**